store_image/utils/func.py

```diff
@@ -0,0 +1,171 @@
+import pymysql
+from sshtunnel import SSHTunnelForwarder
+import asyncio, re
+from playwright.async_api import async_playwright
+import time
+
+def connect_db():
+    tunnel = SSHTunnelForwarder(
+        ('_________', 22),
+        ssh_username="________",
+        ssh_pkey='_______________',
+        remote_bind_address=('_____________', )
+        )
+    
+    tunnel.start()
+
+    db = pymysql.connect(
+        host='127.0.0.1',
+        user="______",
+        password="__________",
+        port=tunnel.local_bind_port,
+        database='____________'
+    )
+    
+    return db, tunnel  
+
+def execute_query(db, sql):
+    try:
+        with db.cursor() as cur:
+            cur.execute(sql)
+            results = cur.fetchall()
+            return results
+    except Exception as e:
+        print(f"Error occurred during query execution: {e}")
+        return None
+
+
+def close_db_connection(db, tunnel):
+    db.close()
+    tunnel.stop()
+
+
+def get_data():
+    results = []
+    db, tunnel = connect_db()
+    sql = "SELECT restaurant_id, name FROM restaurant"
+    results = execute_query(db, sql)    
+    close_db_connection(db, tunnel)
+    
+    return results
+
+
+# 메인 함수에서 이들을 사용하는 방식
+def delete_and_save_data(df_list):
+    result = ""
+    db, tunnel = connect_db()
+
+    delete_query = """
+        DELETE FROM restaurant_image WHERE restaurant_id = %s;
+    """
+
+    insert_query = """
+        INSERT INTO restaurant_image (restaurant_id, image_url)
+        VALUES (%s, %s)
+    """
```

Comment:
p2; 주석은 의미가 없는 내용이면 제거해도 될 것 같습니다!

fyi; in 조건에는 조건 수 제한이 있는데 지금은 데이터가 많지 않아서 괜찮지만 나중에 대용량 데이터를 다룰 경험이 있다면 참고해두면 좋을 것 같아용
> The number of values in the IN() list is only limited by the [max_allowed_packet](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_allowed_packet) value.

-> [mysql docs](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_in)

```suggestion
def delete_and_save_data(df_list):
    result = ""
    db, tunnel = connect_db()

    delete_query = """
        DELETE FROM restaurant_image WHERE restaurant_id = %s;
    """

    insert_query = """
        INSERT INTO restaurant_image (restaurant_id, image_url)
        VALUES (%s, %s)
    """
```

Reply:
해결 :)